gc/base/standard/StandardWriteBarrier.hpp

```diff
@@ -62,7 +63,7 @@ standardWriteBarrier(OMR_VMThread *omrThread, omrobjectptr_t parentObject, omrob
 	}
 #endif /* defined(OMR_GC_MODRON_SCAVENGER) */
 #if defined(OMR_GC_MODRON_CONCURRENT_MARK)
-	if (extensions->concurrentMark) {
+	if ((extensions->concurrentMark) && (extensions->configuration->isIncrementalUpdateBarrierEnabled())) {
```

Comment:
Just a general note (speaking for benefits of OMR), we should introduce some flags about barrier awareness in OMR_VMThread, like we have J9_PRIVATE_FLAGS_CONCURRENT_MARK_ACTIVE in OpenJ9.

In fact we would move them from OpenJ9 to OMR.

I can see some attempts for generational barrier, but the work does not seem to be completed:

```
typedef struct OMR_VMThread {
...
	/* todo: dagar these are temporarily duplicated and should be removed from J9VMThread */
	void *lowTenureAddress;
	void *highTenureAddress;

	void *heapBaseForBarrierRange0;
	uintptr_t heapSizeForBarrierRange0;
```

Reply:
[Opened new Issue for this](https://github.com/eclipse/omr/issues/6258)